include/hal/library/responder/key_pair_info.h

```diff
@@ -0,0 +1,50 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef RESPONDER_KEY_PAIR_INFO_H
+#define RESPONDER_KEY_PAIR_INFO_H
+
+#include "hal/base.h"
+#include "internal/libspdm_lib_config.h"
+#include "industry_standard/spdm.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_GET_KEY_PAIR_INFO_CAP
+
+/**
+ * read the key pair info of the key_pair_id.
+ *
+ * @param  spdm_context               A pointer to the SPDM context.
+ * @param  key_pair_id                Indicate which key pair ID's information to retrieve.
+ *
+ * @param  total_key_pairs            Indicate the total number of key pairs on the responder.
```

Comment:
As was clarified in https://github.com/DMTF/SPDM-WG/issues/3526#issuecomment-2233350803, `total_key_pairs` is fixed. As such it can be stored in the `spdm_context` and removed from this function.

Reply:
Hi. The `total_key_pair` is still returned in the function.
Because the key pair info is provisioned by the responder device, the key pair number should be consistent.

Reply:
We'll see what @jyao1's opinion is when he gets back.

Reply:
Can we design a new API named "libspdm_get_total_key_pairs()" ?

Reply:
>Can we design a new API named libspdm_get_total_key_pairs() ?

If it's going to be returned in a function then it should be in `libspdm_read_key_pair_info`. What I, and the SPDM specification, is saying is that since `total_key_pairs` is fixed at the beginning of the connection and cannot change then just put it in the `spdm_context`.

Reply:
I am fine with that. We can add LIBSPDM_DATA_TOTAL_KEY_PAIRS data type.

Reply:
Just `LIBSPDM_DATA_TOTAL_KEY_PAIRS`.

Reply:
> Just LIBSPDM_DATA_TOTAL_KEY_PAIRS.

This will cause misunderstanding. `LIBSPDM_DATA_TOTAL_KEY_PAIRS` may be interpreted as total key pairs information.

Reply:
Thanks. I have fixed the feedback. Please review the comparing:
https://github.com/DMTF/libspdm/compare/6fc54f2033535b9f78a1bfe0faf519e5c9396ba3..6a04b803a95ec3406e09a6d13e81e1530012a14c